image/templates/compute-sled/targets.json5

```
@@ -0,0 +1,10 @@
+{
+    "gimlet": {
+        "efs": "milan-gimlet-b.efs.json5",
+        "firmware": ["GN/1.0.0.a"],
+    },
+    "cosmo": {
+        "efs": "turin-cosmo-a.efs.json5",
+        "firmware": ["BRH/1.0.0.3-p1"],
+    }
+}
```

Let's make this a TOML file; e.g.,

```suggestion
[gimlet]
efs = "milan-gimlet-b.efs.json5"
firmware = ["GN/1.0.0.a"]

[cosmo]
efs = "turin-cosmo-a.efs.json5"
firmware = ["BRH/1.0.0.3-p1"]
```

---

So at this point we're pretty close to recreating the apps that are consumed by `xtask gen`. :/

---

Is there a document that describes how to use that, and is it a Committed interface etc?

---

Fair enough (on the file format). It's in a directory full of JSON files that are consumed by image builder which is why I used JSON but I'd already diverged slightly to JSON5.

On using xtask gen, I can look at that in a followup commit so we can see how it looks. It probably makes sense although the input files will all need to come from this repo and not ahib, but I can copy the app snapshot across.
The only thing we'll lose as far as I can see is the ability to specify multiple blob directories but I don't know how important that is/was (we don't use it today as far as I know).

---

FWIW, JSON is my largest regret with the image-builder, and I hope to find time to redo it with KDL 2.0 now that that's out and stable!

---

To make AHIB's `xtask gen` usable for this I think we need to add an option to override the automatic configuration file path. It's currently expected to be under `etc/` relative to the ahib directory and named following the app file. That's fine in itself but we need a way to override that path to point into the helios checkout.